packages/translated-components/src/index.js

```diff
@@ -61,17 +62,33 @@ const moneyFormat = (language) => ({
 
 const preHeat = (translations, format) => (
   reduce(translations, (acc, o, language) => {
-    acc[language] = reduce(o, (acc, v, k) => {
-      acc[k] = new IntlFormat(v, kebabCase(language), {...moneyFormat(language), ...format})
-      return acc
-    }, {})
+    acc[language] = preheatValues(o, format, language)
     return acc
   }, {})
 )
 
-const templateReducer = (values) => (acc, v, k) => {
-  acc[k] = v.format(values)
-  return acc
+const preheatValues = (obj, format, language) => {
+  return reduce(obj, (acc, value, key) => {
+    if (typeof value === 'object') {
+      acc[key] = preheatValues(value, format, language)
+    } else {
+      acc[key] = new IntlFormat(value, kebabCase(language), {...moneyFormat(language), ...format})
+    }
+    return acc
+  }, {})
+}
+
+const templateReducer = (values) => partial(templateIteratee, values)
```

Comment:
I was suggesting what ben is suggesting. If the signature of your function is:

```
Array -> function
```

Then that's cool. I'd prefer that. I _really_ don't like seeing `bind` used for partial application. 

Like I'd expect to see partial if you were using an existing function from a lib that you wanted to pass to `reduce` or `map` etc... but in this case your defining it and using it right there. No need to add the indirection of `partial`, in my opinion anyway.